saleor/graphql/product/types/products.py

```diff
@@ -8,28 +7,40 @@
 from graphql.error import GraphQLError
 from graphql_jwt.decorators import permission_required
 
-from ...product import models
-from ...product.templatetags.product_images import (
+from ....product import models
+from ....product.templatetags.product_images import (
     get_product_image_thumbnail, get_thumbnail)
-from ...product.utils import calculate_revenue_for_variant
-from ...product.utils.availability import get_availability
-from ...product.utils.costs import (
+from ....product.utils import calculate_revenue_for_variant
+from ....product.utils.availability import get_availability
+from ....product.utils.costs import (
     get_margin_for_variant, get_product_costs_data)
-from ..core.connection import CountableDjangoObjectType
-from ..core.enums import ReportingPeriod, TaxRateType
-from ..core.fields import PrefetchingConnectionField
-from ..core.types import Image, Money, MoneyRange, TaxedMoney, TaxedMoneyRange
-from ..translations.enums import LanguageCodeEnum
-from ..translations.resolvers import resolve_translation
-from ..translations.types import (
-    AttributeTranslation, AttributeValueTranslation, CategoryTranslation,
-    CollectionTranslation, ProductTranslation, ProductVariantTranslation)
-from ..utils import get_database_id, reporting_period_to_date
-from .descriptions import AttributeDescriptions, AttributeValueDescriptions
-from .enums import AttributeValueType, OrderDirection, ProductOrderField
-
-COLOR_PATTERN = r'^(#[0-9a-fA-F]{3}|#(?:[0-9a-fA-F]{2}){2,4}|(rgb|hsl)a?\((-?\d+%?[,\s]+){2,3}\s*[\d\.]+%?\))$'  # noqa
-color_pattern = re.compile(COLOR_PATTERN)
+from ...core.connection import CountableDjangoObjectType
+from ...core.enums import ReportingPeriod, TaxRateType
+from ...core.fields import PrefetchingConnectionField
+from ...core.types import Image, Money, MoneyRange, TaxedMoney, TaxedMoneyRange
+from ...translations.enums import LanguageCodeEnum
+from ...translations.resolvers import resolve_translation
+from ...translations.types import (
+    CategoryTranslation, CollectionTranslation, ProductTranslation,
+    ProductVariantTranslation)
+from ...utils import get_database_id, reporting_period_to_date
+from ..enums import OrderDirection, ProductOrderField
+from .attributes import Attribute, SelectedAttribute
+from .digital_contents import DigitalContent
+
+
+def prefetch_products(info, *_args, **_kwargs):
+    """Prefetch products visible to the current user.
+
+    Can be used with models that have the `products` relationship. Queryset of
+    products being prefetched is filtered based on permissions of the viewing
```

Comment:
Sounds weird to me, what do you think?
```suggestion
    products being prefetched is filtered based on permissions of the requesting
```